src/app_service/service/community_tokens/async_tasks.nim

```diff
@@ -227,20 +236,86 @@ type
 const fetchCollectibleOwnersTaskArg: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
   let arg = decode[FetchCollectibleOwnersArg](argEncoded)
   try:
-    let response = collectibles.getCollectibleOwnersByContractAddress(arg.chainId, arg.contractAddress)
+    var response = collectibles.getCollectibleOwnersByContractAddress(arg.chainId, arg.contractAddress)
+
+    var owners = fromJson(response.result, CollectibleContractOwnership).owners
+    owners = owners.filter(x => x.address != ZERO_ADDRESS)
+
+    response = communities_backend.getCommunityMembersForWalletAddresses(arg.communityId, arg.chainId)
+
+    let communityCollectibleOwners = owners.map(proc(owner: CollectibleOwner): CommunityCollectibleOwner =
+      let ownerAddressUp = owner.address.toUpper()
+      for responseAddress in response.result.keys():
+        let responseAddressUp = responseAddress.toUpper()
+        if ownerAddressUp == responseAddressUp:
+          let member = response.result[responseAddress].toContactsDto()
+          return CommunityCollectibleOwner(
+            contactId: member.id,
+            name: member.displayName,
+            imageSource: member.image.thumbnail,
+            collectibleOwner: owner
+          )
+      return CommunityCollectibleOwner(collectibleOwner: owner)
+    )
 
-    if not response.error.isNil:
-      raise newException(ValueError, "Error getCollectibleOwnersByContractAddress" & response.error.message)
+    let output = %* {
+      "chainId": arg.chainId,
+      "contractAddress": arg.contractAddress,
+      "communityId": arg.communityId,
+      "result": %communityCollectibleOwners,
+      "error": ""
+    }
+    arg.finish(output)
+  except Exception as e:
+    let output = %* {
+      "chainId": arg.chainId,
+      "contractAddress": arg.contractAddress,
+      "communityId": arg.communityId,
+      "result": "",
+      "error": e.msg
+    }
+    arg.finish(output)
+
+type
+  FetchAssetOwnersArg = ref object of QObjectTaskArg
+    chainId*: int
+    contractAddress*: string
+    communityId*: string
+
+const fetchAssetOwnersTaskArg: Task = proc(argEncoded: string) {.gcsafe, nimcall.} =
+  let arg = decode[FetchAssetOwnersArg](argEncoded)
+  try:
+    let addressesResponse = communities_backend.getCommunityMembersForWalletAddresses(arg.communityId, arg.chainId)
+    var allCommunityMembersAddresses: seq[string] = @[]
+    for address in addressesResponse.result.keys():
+      allCommunityMembersAddresses.add(address)
+
+    let balancesResponse = backend.getBalancesByChain(@[arg.chainId], allCommunityMembersAddresses, @[arg.contractAddress])
```

Comment:
I also use this endpoint, maybe we can generalise nim parsing of this response? For my the case, 'seq[WalletTokenDto]' should be fine